GVRf/Framework/framework/src/main/java/org/gearvrf/GVRAndroidResource.java

```diff
@@ -313,6 +313,23 @@ public synchronized final InputStream getStream() throws IOException{
         return stream;
     }
 
+    /**
+     * Sets the stream for a resource.
+     * This function allows you to provide a stream that is already open to
+     * an existing resource. It will throw an exception if that resource
+     * already has an open stream.
+     * @param s InputStream currently open stream to use for I/O.
+     */
+    protected synchronized void setStream(InputStream s)
```

Comment:
GVRAndroidResource already has a constructor that accepts an input stream - look at line 208. Might have to make it public to access it.